docs/javascript-unit-tests.md

```diff
@@ -0,0 +1,712 @@
+   } );
+   ```
+1. Run your sample test using `gulp test:unit --specs=js/modules/sample-spec.js`
+   (substituting your own filename).
+   The test should fail – this is expected.
+   Remember, when doing TDD, we want to write our test to fail first,
+   then write the corresponding JavaScript that will make the test pass.
+1. Make the test pass by changing your script’s line 7
+   ([see `sample.js`](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/unprocessed/js/modules/sample.js))
+   to the following:
+   ```js
+   return 'Shredder';
+   ```
+1. Run the test again to confirm the test now passes.
+   Doesn’t it feel good?
+
+[See the “Running tests” section of this page](#running-tests)
+for additional commands to run tests.
+
+
+### File structure (basic layout of a test file)
+
+In order to make the `sample-spec.js` more meaningful to your own use case,
+you’ll need to know how to structure a unit test using Jest methods.
+Let’s take a look at the structure of our very basic sample test file.
+
+#### Loading dependencies
+
+Line 1 of any spec file will use an `import` statement to
+include the JavaScript file that you are testing.
+Additional dependencies should be added in the same manner.
+
+```js
+import * as sample from '../../../../cfgov/unprocessed/js/modules/sample.js';
+```
+
+Some test files use `const` declarations to
+`require` scripts instead of `import`,
+because those files were written before `import` was available.
+We prefer to use `import` because it allows for
+[tree shaking in Webpack](https://webpack.js.org/guides/tree-shaking/),
+meaning if two modules are importing the same module
+it should only be included in the bundle once,
+whereas with `require` it would be included twice.
+
+A consequence is that variables can't be used in the import path,
+as they prevent Webpack from figuring out which modules are duplicates.
+For example, this snippet shows how a `require` statement
+should be converted to an `import` statement,
+but without including the `BASE_JS_PATH` variable in the file path:
+
+```js
+// works but could duplicate footer-button.js if other files also require it
+const FooterButton = require( BASE_JS_PATH + 'modules/footer-button.js' );
+
+// doesn't work and the build will fail
+import * as FooterButton from BASE_JS_PATH + 'modules/footer-button.js';
+
+// is ugly but it works and supports tree shaking
+import * as FooterButton from '../../../../cfgov/unprocessed/js/modules/footer-button.js';
+```
+
+`import` also provides a benefit in that you can choose specific parts
+of a module so that you only import the dependencies you need.
+For testing purposes, we will typically import the whole module
+to make sure we have full test coverage.
+[Read the `import` reference guide on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import)
+on how to implement `import` for different use cases.
+
+#### The `describe` function
+
+In Jest (whose syntax is based on Jasmine),
```

Comment:
Link to [Jasmine docs](https://jasmine.github.io/) here, as this is the first time it's being referenced in this doc?

Reply:
I don't think so; it's just a historical note. Wouldn't want anyone actually referring to the Jasmine docs.